pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -115,15 +116,20 @@ func (d *Drift) isDrifted(ctx context.Context, nodePool *v1beta1.NodePool, nodeC
 	return driftedReason, nil
 }
 
-// Eligible fields for static drift are described in the docs
+// Eligible fields for drift are described in the docs
 // https://karpenter.sh/docs/concepts/deprovisioning/#drift
 func areStaticFieldsDrifted(nodePool *v1beta1.NodePool, nodeClaim *v1beta1.NodeClaim) cloudprovider.DriftReason {
 	nodePoolHash, foundHashNodePool := nodePool.Annotations[v1beta1.NodePoolHashAnnotationKey]
 	nodeClaimHash, foundHashNodeClaim := nodeClaim.Annotations[v1beta1.NodePoolHashAnnotationKey]
 	if !foundHashNodePool || !foundHashNodeClaim {
 		return ""
 	}
-	return lo.Ternary(nodePoolHash != nodeClaimHash, NodePoolDrifted, "")
+	// validate that the version of the crd is the same
+	nodePoolHashSlice, nodeClaimHashSlice := strings.Split(nodePoolHash, "-"), strings.Split(nodeClaimHash, "-")
```

Comment:
Are we using the version annotation or are we prefixing with the version? This indicates that we are prefixing with the version, but from looking at the annotations that you have listed in the other file, it looks like we have a separate annotation for it

Reply:
The goal will be to use a version annotation, to avoid a string splitting.